File: two_pointers/remove_duplicates_from_sorted_array.js
// Q:26 - https://leetcode.com/problems/remove-duplicates-from-sorted-array/

/**
 * @param {number[]} nums
 * @return {number}
 */
 var removeDuplicates = function(nums) {
    let firstNonDuplicateIndex = 1;
    
    for (let i = 1; i < nums.length; i++){
        if (nums[i] === nums[i - 1]){
            continue;
        }
                
        nums[firstNonDuplicateIndex] = nums[i];
        firstNonDuplicateIndex += 1;
    }
        
    return firstNonDuplicateIndex;
};

File: two_pointers/remove_duplicates_from_sorted_array.py
# Q:26 - https://leetcode.com/problems/remove-duplicates-from-sorted-array/

from typing import List

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        lastIndex = 1

        for index in range(1, len(nums)):
            if nums[index - 1] != nums[index]:
                nums[lastIndex] = nums[index]
                lastIndex += 1
        
        return lastIndex